pkg/ifuzz/ifuzz_test.go

```diff
@@ -116,6 +116,46 @@ func testDecode(t *testing.T, arch string) {
 	}
 }
 
+func TestGenerate(t *testing.T) {
+	for _, arch := range allArches {
+		t.Run(arch, func(t *testing.T) {
+			testGenerate(t, arch)
+		})
+	}
+}
+
+func testGenerate(t *testing.T, arch string) {
+	insnset := iset.Arches[arch]
+	r := rand.New(testutil.RandSource(t))
+	for mode := iset.Mode(0); mode < iset.ModeLast; mode++ {
+		for repeat := 1; repeat < 10; repeat++ {
+			if len(insnset.GetInsns(mode, iset.TypeUser)) == 0 {
+				continue
+			}
+			cfg := &iset.Config{
+				Arch: arch,
+				Mode: mode,
+				Priv: true,
+				Exec: true,
+				Len:  repeat,
+			}
+			text := Generate(cfg, r)
+			pos := 0
```

Comment:
delete this and size

Reply:
Oh, sorry

Reply:
Interesting, `insnset.Decode()` may return `0, nil` on x86, is that expected?

Reply:
I don't remember exactly. It should be mainly used during mutation when we try to split code into instructions and mutate them separately. Is that code handles `0, nil` well, then we are good.

Reply:
I made the loop in the test check for both size==0 and err!=nil to account for this.

Reply:
But also Generate shouldn't return bad instruction IIRC, so decoding of them shouldn't produce `0, nil` I think. I can imagine `0, nil` (decoding error) happens for already mutated instructoins.